### PR TITLE
add interface to allow default fields into formatter

### LINKED ISSUE #10: Allow to add custom fields into JsonStorage
Hi, I'm trying to use `tracing` library to emit json line style logs, and found this library. It looks very nice.
In our project, we want to add some custom fields into all logs as default fields. So, I added another layer to add those fields into `JsonStorage` after `JsonStorageLayer`, but found that we cannot add them directly because `JsonStorage` doesn't have a function to provide mutable reference for inner `HashMap`.
So, I'd like to suggest to add a new function to provide mutable reference for inner map (something like `values_mut()` or adding `DerefMut` implementation), or allow to add custom default fields when constructing `JsonStorageLayer`, as `createLogger` in [node-bunyan](https://github.com/trentm/node-bunyan) does.

I can open a PR if you think these approaches are reasonable. I'm not familiar with `tracing`, so I'm not confident with them, so please tell me if you have any other ways to achieve them.

### PR DESCRIPTION
**Pull request by garasubo**
Fix #10 by allowing default fields when constructing the formatter

### CODE REVIEW

**Comment by garasubo** [comment]
@LukeMathWalker Hi Luke, do you have any chance to review this PR? Thank you.

File: src/formatting_layer.rs [hunk 1]
```diff
@@ -75,6 +76,18 @@ impl<W: for<'a> MakeWriter<'a> + 'static> BunyanFormattingLayer<W> {
             pid: std::process::id(),
             hostname: gethostname::gethostname().to_string_lossy().into_owned(),
             bunyan_version: 0,
+            default_fields: Vec::new(),
+        }
+    }
+
+    pub fn with_default_fields(name: String, make_writer: W, default_fields: Vec<(String, Value)>) -> Self {
```
Comment (LukeMathWalker) [inline src/formatting_layer.rs:83]: I'd prefer to use `HashMap` here instead of `Vec` given that we don't want repeated keys.

**Comment by LukeMathWalker** [comment]
Just one comment, the rest looks 👌🏻

File: src/formatting_layer.rs [hunk 2]
```diff
@@ -75,6 +76,18 @@ impl<W: for<'a> MakeWriter<'a> + 'static> BunyanFormattingLayer<W> {
             pid: std::process::id(),
             hostname: gethostname::gethostname().to_string_lossy().into_owned(),
             bunyan_version: 0,
+            default_fields: Vec::new(),
```
Comment (LukeMathWalker) [inline src/formatting_layer.rs:79]: `new` should call `with_default_fields` internally passing an empty set of fields - we don't want to duplicate the initialisation logic for the other fields.

File: src/formatting_layer.rs (hunk 1 above)
Comment (garasubo) [reply src/formatting_layer.rs]: Fixed

File: src/formatting_layer.rs (hunk 2 above)
Comment (garasubo) [reply src/formatting_layer.rs]: Fixed

**Comment by LukeMathWalker** [comment]
Thanks!

**Pull request merged**